packages/core/src/codewhisperer/util/zipUtil.ts

```diff
@@ -61,8 +72,61 @@ export class ZipUtil {
         return content
     }
 
-    public reachSizeLimit(size: number): boolean {
-        return size > this.getFileScanPayloadSizeLimitInBytes()
+    public isJavaClassFile(uri: vscode.Uri) {
+        return uri.fsPath.endsWith(ZipConstants.javaBuildExt)
+    }
+
+    public reachSizeLimit(size: number, scanType: CodeWhispererConstants.SecurityScanType): boolean {
+        if (scanType === CodeWhispererConstants.SecurityScanType.File) {
+            return size > this.getFileScanPayloadSizeLimitInBytes()
+        } else {
+            return size > this.getProjectScanPayloadSizeLimitInBytes()
+        }
+    }
+
+    public willReachSizeLimit(current: number, adding: number): boolean {
+        const willReachLimit = current + adding > this.getProjectScanPayloadSizeLimitInBytes()
+        return willReachLimit
+    }
+
+    protected async traverseDir(dirPath: string, ignore?: Ignore) {
```

Comment:
This is doing a lot more than "traverse dir". What is it actually doing? The docstring should say so. Or use a more meaningfull name, such as `collectFiles`
```suggestion
    /** Recursively add files to ... and compute stats ... */
    protected async collectFiles(dirPath: string, ignore?: Ignore) {
```

Please, make this a habit for all non-obvious code in the future.